backend/accounts/notes/notes_test.go

```diff
@@ -68,3 +77,26 @@ func TestMaxLen(t *testing.T) {
 	require.NoError(t, notes.SetTxNote("tx-id", strings.Repeat("x", 1024)))
 	require.Error(t, notes.SetTxNote("tx-id", strings.Repeat("x", 1025)))
 }
+
+func TestMergeLegacy(t *testing.T) {
+	notes, err := LoadNotes(test.TstTempFile("account-notes"))
+	require.NoError(t, err)
+	require.NoError(t, notes.SetTxNote("tx-id-1", "note for tx-id-1"))
+	require.NoError(t, notes.SetTxNote("tx-id-2", "note for tx-id-2"))
+
+	legacyNotes, err := LoadNotes(test.TstTempFile("legacy-notes"))
+	require.NoError(t, err)
+	require.NoError(t, legacyNotes.SetTxNote("tx-id-1", "legacy note for tx-id-1"))
+	require.NoError(t, legacyNotes.SetTxNote("tx-id-3", "legacy note for tx-id-3"))
+
+	require.NoError(t, notes.MergeLegacy(legacyNotes))
+	require.Equal(t,
+		&Data{
+			TransactionNotes: map[string]string{
+				"tx-id-1": "note for tx-id-1",
+				"tx-id-2": "note for tx-id-2",
+				"tx-id-3": "legacy note for tx-id-3",
+			},
+		},
+		notes.Data())
+}
```

Comment:
Should we check that the merged notes have been persisted in the file?

Reply:
Done